examples/Thermostat/Server/ThermoBridge.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) ZeroC, Inc.
+
+using IceRpc.Features;
+using IceRpc.Slice;
+using Igloo;
+using System.Diagnostics;
+
+namespace ThermostatServer;
+
+/// <summary>Implements Slice interface `ThermoHome`.</summary>
+
+internal class ThermoBridge : Service, IThermoHomeService
+{
+    private readonly DeviceConnection _deviceConnection;
+    private readonly ThermoFacade _thermoFacade;
+
+    public ValueTask ReportAsync(
+        IAsyncEnumerable<Reading> readings,
+        IFeatureCollection features,
+        CancellationToken cancellationToken)
+    {
+        IDispatchInformationFeature? dispatchInfo = features.Get<IDispatchInformationFeature>();
+
+        // We installed the dispatch middleware in Router to get this dispatch info:
+        Debug.Assert(dispatchInfo is not null);
+
+        _deviceConnection.SetInvoker(dispatchInfo.ConnectionContext.Invoker);
```

Comment:
I would add a comment here something like:

A device has called home to start reporting readings, we update the device connection invoker, with the invoker from the connection context, this invoker allows to send requests to the just connected device